### DISCUSSION TITLE
DEPRECATED message No Availability for some countries

### DISCUSSION THREAD

**Discussion by Ephigenia** [Ephigenia/ikea-availability-checker#83] [Q&A] [answered]
UPDATE: 

# 2021-12-31

After some more hints by @klim123123 I found out that the DEPRECATION warning appears when product availability is requested for a product that is not available in the country. So the ikea-availability-checker is still working as expected.

# 2021-12-17

At least since 17.12.2021 some IOWS2 endpoints for some countries reply with a deprecation warning ("IOWSStockAvailabilityService.GetStockAvailability.v1 API is now deprecated.") and no data at all. The following countries are known to not provide any data anymore:

- au
- ca
- gb
- hk
- jp
- kr
- my
- pl (only Warszawa)
- ru
- sg
- th
- us

The reply contains the following headers which hint that also other countries may be added to the list in the next future.
```javascript
{
  'x-backside-transport': 'FAIL FAIL,FAIL FAIL,FAIL FAIL',
  'deprecation': 'version="1", date="Sat, 31 Dec 2022 23:59:59 GMT"',
  'warning': 'IOWSStockAvailabilityService.GetStockAvailability.v1 API is now deprecated.',
  'link': 'alternate="Customer Item Availability. Enquire via slack #rrm-cia"',
}
```

The table will display "DEPRECATED" for requests that run into this deprecation error since 1.0.4 (fixed in PR #80

**Comment by Ephigenia** [comment]
Maybe the new developer portal and it’s APIs provide new possibilities: https://github.com/Ephigenia/ikea-availability-checker/discussions/84

**Comment by klim123123** [comment]
ru - works fine. no errors.

**Reply by Ephigenia** [reply]
@klim123123 interesting, I’ve tested with the following product id and command and get the deprecated header:

```
ikea-availability stock --country ru 90208543
```

I’m receiving deprecated headers from all stores:

```
──────────────────────────┬─────────────┬────────────────────┬──────────┬──────────────────┬──────────────────┬───────┬─────────────┬─────────────┬──────────┐
│ date                     │ countryCode │ country            │ product  │ storeId (buCode) │ store            │ stock │ probability │ restockDate │ forecast │
├──────────────────────────┼─────────────┼────────────────────┼──────────┼──────────────────┼──────────────────┼───────┼─────────────┼─────────────┼──────────┤
│ 2021-12-31T08:46:37.285Z │ ru          │ Russian Federation │ 90208543 │ 524              │ Интернет-магазин │     0 │  DEPRECATED │             │          │
└──────────────────────────┴─────────────┴────────────────────┴──────────┴──────────────────┴──────────────────┴───────┴─────────────┴─────────────┴──────────┘
```

Can you verify this or provide the product id / command that you’re using?

**Reply by klim123123** [reply]
I think this DEPRECATED message is about 524 store.
For example, use 426 or 464 store

ikea-availability stock --store=464,426 40482596

**Reply by Ephigenia** [reply]
It looks like the product id makes the difference. When I’m using `40482596` it works fine. Don’t know why they are making a difference in the product.

**Reply by Ephigenia** [reply]
I found out that when I use `40482596` with the stores from Germany I get the DEPRECATED message. So I guess that the deprecation message happens when a product is not available in a country. I was able to confirm my theory by searching in the german store for the product id: https://www.ikea.com/de/de/search/products/?q=40482596 which returns no results. I’ll change the note in the README so that it gets more clear.

**Reply by Ephigenia** [reply]
@klim123123 thanks for your feedback. I’m happy to see that the availability-checker still works :)

**Reply by klim123123** [reply]
Thank for you for your work!

**Reply by jeanf-flutterint** [reply]
I tried https://www.ikea.com/ca/en/search/products/?q=49318323 and got 1 result in North York but 
`ikea-availability stock --store 004,149,372 59227775 29027396 49318323`

$ ikea-availability stock --store 004,149,372 59227775 29027396 49318323
┌──────────────────────────┬─────────────┬─────────┬──────────┬──────────────────┬────────────┬───────┬─────────────┬─────────────┬──────────┐
│ date                     │ countryCode │ country │ product  │ storeId (buCode) │ store      │ stock │ probability │ restockDate │ forecast │
├──────────────────────────┼─────────────┼─────────┼──────────┼──────────────────┼────────────┼───────┼─────────────┼─────────────┼──────────┤
│ 2022-07-05T19:50:14.197Z │ ca          │ Canada  │ 59227775 │ 149              │ North York │     0 │  DEPRECATED │             │          │
├──────────────────────────┼─────────────┼─────────┼──────────┼──────────────────┼────────────┼───────┼─────────────┼─────────────┼──────────┤
│ 2022-07-05T19:50:13.893Z │ ca          │ Canada  │ 59227775 │ 004              │ Ottawa     │     0 │  DEPRECATED │             │          │
├──────────────────────────┼─────────────┼─────────┼──────────┼──────────────────┼────────────┼───────┼─────────────┼─────────────┼──────────┤
│ 2022-07-05T19:50:13.808Z │ ca          │ Canada  │ 59227775 │ 372              │ Vaughan    │     0 │  DEPRECATED │             │          │
├──────────────────────────┼─────────────┼─────────┼──────────┼──────────────────┼────────────┼───────┼─────────────┼─────────────┼──────────┤
│ 2022-07-05T19:50:13.892Z │ ca          │ Canada  │ 29027396 │ 149              │ North York │     0 │  DEPRECATED │             │          │
├──────────────────────────┼─────────────┼─────────┼──────────┼──────────────────┼────────────┼───────┼─────────────┼─────────────┼──────────┤
│ 2022-07-05T19:50:13.802Z │ ca          │ Canada  │ 29027396 │ 004              │ Ottawa     │     0 │  DEPRECATED │             │          │
├──────────────────────────┼─────────────┼─────────┼──────────┼──────────────────┼────────────┼───────┼─────────────┼─────────────┼──────────┤
│ 2022-07-05T19:50:13.891Z │ ca          │ Canada  │ 29027396 │ 372              │ Vaughan    │     0 │  DEPRECATED │             │          │
├──────────────────────────┼─────────────┼─────────┼──────────┼──────────────────┼────────────┼───────┼─────────────┼─────────────┼──────────┤
│ 2022-07-05T19:50:13.918Z │ ca          │ Canada  │ 49318323 │ 149              │ North York │     0 │  DEPRECATED │             │          │
├──────────────────────────┼─────────────┼─────────┼──────────┼──────────────────┼────────────┼───────┼─────────────┼─────────────┼──────────┤
│ 2022-07-05T19:50:14.074Z │ ca          │ Canada  │ 49318323 │ 004              │ Ottawa     │     0 │  DEPRECATED │             │          │
├──────────────────────────┼─────────────┼─────────┼──────────┼──────────────────┼────────────┼───────┼─────────────┼─────────────┼──────────┤
│ 2022-07-05T19:50:13.904Z │ ca          │ Canada  │ 49318323 │ 372              │ Vaughan    │     0 │  DEPRECATED │             │          │
#

**Comment by Ephigenia** [comment, marked as answer]
@jeanpfrancois This issue is fixed by using a different API. This "new" API is used by the alpha channel only. Please read: https://github.com/Ephigenia/ikea-availability-checker#important-note

For your use-case please try:

```
npx ikea-availability-checker@2.0.0-alpha.3 stock --store 004,149,372 59227775 29027396 49318323
```

**Reply by jeanf-flutterint** [reply]
Yes, it works! Thank you!

**Reply by jeanf-flutterint** [reply]
I suggest adding CA in this list:
`This package uses the undocumented iows2 api by IKEA which doesn’t work for several countries anymore: be, cn, fi, fr, gb, hk, ie, jp, kr, my, pl, ru, sg, th, tw, us`
